src/main/java/com/mael/payroll/Employee.java

```diff
@@ -60,7 +60,7 @@ public PaymentMethod getPaymentMethod() {
     }
 
     public void getPaid(Paycheck paycheck) {
-        double netPay = getPaymentType().calculatePay();
+        double netPay = getPaymentType().calculatePay(paycheck.getDate());
```

Comment:
Seeing that `getPaid()` only mutates its argument feels weird at first.
What should be expected when an employee actually _gets paid_? I mean it from an employee behavioural point of view. As an employee, when I get paid, do I modify my paycheck?

I can't quite put my finger on it, but it seems to me that some abstraction or dependency is upside down when it comes to the paychecks?
Can you think of another way of paying an employee?